### ISSUE TITLE
No 2.33-0ubuntu9.2 package for ubuntu 20.04

### ISSUE THREAD

**Issue by KomeilParseh** [o2sh/onefetch#458]
# No 2.33-0ubuntu9.2 package for ubuntu 20.04

Hi, thank you very much for your app

I had already installed it in the **arch** and it worked without any problems, but unfortunately now that I want to install it on **Ubuntu 20.04**, the installer says that the prerequisite for the program on your operating system is old.

Can this program (prerequisite) be reduced to a few previous versions?

Thanks 🌹

## apt error message
```
...

The following packages have unmet dependencies:
 onefetch : Depends: libc6 (>= 2.33) but 2.31-0ubuntu9.2 is to be installed
E: Unable to correct problems, you have held broken packages.
```

**Comment by o2sh** [comment]
I tried to build a new package for ubuntu 20.04 but it failed (see [buildlog](https://launchpadlibrarian.net/547205174/buildlog_ubuntu-focal-amd64.onefetch_2.10.2-focal_BUILDING.txt.gz)).

The culprit being the `bool::then` syntax that we started using in v2.10.0 and that was stabilized by the Rust compiler in [1.50.0](https://blog.rust-lang.org/2021/02/11/Rust-1.50.0.html). Unfortunately, this version of rustc is only available in [Hirsute](https://launchpad.net/ubuntu/+source/rustc).

Maybe you could try installing Onefetch with [snap](https://github.com/o2sh/onefetch/wiki/Installation#snap) instead.

Very sorry about that :disappointed:

**Comment by spenserblack** [comment]
>`bool::then`

That's my bad, I shouldn't have suggested it without checking the Rust version where it was stabilized.
Should we look into targeting older Rust versions?

**Comment by o2sh** [comment]
I don't think we should refrain from using new syntax - as long as it's stable (ci). 
I mean people will/should upgrade eventually...:thinking:

**Comment by KomeilParseh** [comment]
> I don't think we should refrain from using new syntax - as long as it's stable (ci).
> I mean people will/should upgrade eventually...🤔

Hi, the fact is that many operating systems have not yet been updated to this version, such as dabian, Ubuntu 20.04 and others that many people use these operating systems, so if it is really possible to rewrite it, which is great, otherwise So wait for the update to a newer version or change the operating system!

**Comment by chengleqi** [comment]
I‘ ve got the same error on Ubuntu 20.04

**Comment by o2sh** [comment]
Since v1.51 of rustc is now available to ubuntu version >= 18.04, I was able to build and deploy  [onefetch v2.10.2](https://launchpad.net/~o2sh/+archive/ubuntu/onefetch/+packages) for **Ubuntu 20.04**.

To install onefetch from launchpad:

```bash
sudo add-apt-repository ppa:o2sh/onefetch
sudo apt-get update
sudo apt-get install onefetch
```

Hopefully, this should resolve this issue.

**Comment by chengleqi** [comment]
> Since v1.51 of rustc is now available to ubuntu version >= 18.04, I was able to build and deploy [onefetch v2.10.2](https://launchpad.net/~o2sh/+archive/ubuntu/onefetch/+packages) for **Ubuntu 20.04**.
> 
> To install onefetch via launchpad:
> 
> ```shell
> sudo add-apt-repository ppa:o2sh/onefetch
> sudo apt-get update
> sudo apt-get install onefetch
> ```
> 
> Hopefully, this should resolve this issue.

It works on **Ubuntu 20.04**, thank u!

**Comment by VarLad** [comment]
@o2sh Sorry for messaging here, but I've added the instructions for Fedora in the wiki
I think it just got added, I thought there would something like a PR for that

Either way, you can take a look and see if something is wrong

**Comment by o2sh** [comment]
@VarLad It's perfect, thanks a lot for your contribution 👍
